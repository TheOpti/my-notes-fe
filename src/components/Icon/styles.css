.root:before {
  font-family: "fontello";
  font-style: normal;
  font-weight: normal;
  text-decoration: inherit;
  text-align: center;
  font-variant: normal;
  text-transform: none;
  line-height: 1em;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

/* Sizing */
.small { font-size: 16px; } 
.medium { font-size: 24px; } 
.large { font-size: 32px; }

/* Icons definitions */
.th:before { content: '\e800'; }
.user:before { content: '\e801'; }
.ok:before { content: '\e802'; }
.arrows-cw:before { content: '\e803'; }
.th-list:before { content: '\e804'; }
.tag:before { content: '\e805'; }
.pencil:before { content: '\e806'; }
.trash-empty:before { content: '\e807'; }
.cancel:before { content: '\e808'; }
.pin:before { content: '\e809'; }
.bell:before { content: '\e80a'; }
.cog:before { content: '\e80b'; }
.archive-1:before { content: '\e80c'; }
.archive-2:before { content: '\e80d'; }
.box:before { content: '\e80e'; }
.palette:before { content: '\e80f'; }
.search:before { content: '\e810'; }
.menu:before { content: '\f0c9'; }
.lightbulb:before { content: '\f0eb'; }
.upload-cloud:before { content: '\f0ee'; }
.folder-open-empty:before { content: '\f115'; }
.file-archive:before { content: '\f1c6'; }