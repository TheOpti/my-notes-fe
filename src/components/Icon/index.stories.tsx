import * as React from 'react';
import { storiesOf } from '@storybook/react'

import styles from './styles.css';
import Icon from './'

const { root, small, medium, large, ...rest } = styles;

storiesOf('Icon', module)
  .add('All icons', () => (
    <div style={{ display: 'flex', flexWrap: 'wrap' }}>
      { Object.keys(rest).map((iconName) => (
        <div style={{ margin: '20px 10px', textAlign: 'center', width: '80px' }}> 
          <Icon name={iconName} />
          <p>{iconName}</p>
        </div>
      ))}
    </div>
  ))
  .add('With sizing', () => (
    <div style={{ display: 'flex' }}>
      <div style={{ margin: '20px 10px', textAlign: 'center', width: '80px' }}> 
        <Icon name="tag" size="small" />
        <p>Small</p>
      </div>
      <div style={{ margin: '20px 10px', textAlign: 'center', width: '80px' }}> 
        <Icon name="tag" size="medium" />
        <p>Medium</p>
      </div>
      <div style={{ margin: '20px 10px', textAlign: 'center', width: '80px' }}> 
        <Icon name="tag" size="large" />
        <p>Large</p>
      </div>
    </div>
  ));