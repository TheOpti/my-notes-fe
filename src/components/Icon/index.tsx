import React from 'react';
import cx from 'classnames';
import styles from './styles.css';

type IconPropsType = {
	name: string;
	size?: string;
	className?: string;
};
const Icon: React.FC<IconPropsType> = (props: IconPropsType) => {
	const { name, size, className } = props;

	const iconClasses = cx(styles.root, styles[size], styles[name], className);

	return <i className={iconClasses} data-testid={`icon-${name}`} />;
};

Icon.defaultProps = {
	size: 'medium',
	className: '',
};

export default Icon;
