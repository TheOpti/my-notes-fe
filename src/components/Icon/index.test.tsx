import React from 'react';
import { mount } from 'enzyme';
import Icon from './';

describe('Icon', () => {
	it('should render correctly without any issues', () => {
		const wrapper = mount(<Icon size="small" name="pencil" />);

		expect(wrapper).toMatchSnapshot();
	});
});
