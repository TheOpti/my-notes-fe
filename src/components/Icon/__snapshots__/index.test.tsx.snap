// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should render correctly without any issues 1`] = `
<Icon
  className=""
  name="pencil"
  size="small"
>
  <i
    className="root small pencil"
    data-testid="icon-pencil"
  />
</Icon>
`;
